src/UCTSearch.cpp

```diff
@@ -780,6 +786,11 @@ int UCTSearch::think(int color, passflag_t passflag) {
 }
 
 void UCTSearch::ponder() {
+    bool disable_reuse = !cfg_analyze_tags.m_moves_to_avoid.empty();
+    if (disable_reuse) {
```

Comment:
This will disable tree reuse, but the information in the NNCache will still be incorrect unless the tags are included in the hash, or the cache is cleared here.

Reply:
Disregard this, I forgot that our cache doesn't store the tree search outcome, just the neural network output (it's not a transposition table).